### PR TITLE
[BR] Upgrade bazel remote runner to use ubuntu 20.04 image

### PR DESCRIPTION
**Pull request by maggie-lou**
Seeing `GLIBC_2.28' not found` build errors on the 18.04 image

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5428** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/5428?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/5428?utm_source=stack-comment).

**Review by tylerwilliams — APPROVED** [review]
I don't think anyone is using bazel remote atm so this seems fine, assuming that's all it will affect.

Would it make sense to have remote bazel use the defaults from the buildbuddy.yaml file in the repository? (and fall back to the default buildbuddy.yaml if there is none)

**Comment by maggie-lou** [comment]
> I don't think anyone is using bazel remote atm so this seems fine, assuming that's all it will affect.
> 
> Would it make sense to have remote bazel use the defaults from the buildbuddy.yaml file in the repository? (and fall back to the default buildbuddy.yaml if there is none)

People can specify a container image for each action in buildbuddy.yaml, so it might get complicated to pick a single default. I added a flag to remote bazel to specify image instead

**Comment by tylerwilliams** [comment]
> > I don't think anyone is using bazel remote atm so this seems fine, assuming that's all it will affect.
> > Would it make sense to have remote bazel use the defaults from the buildbuddy.yaml file in the repository? (and fall back to the default buildbuddy.yaml if there is none)
> 
> People can specify a container image for each action in buildbuddy.yaml, so it might get complicated to pick a single default. I added a flag to remote bazel to specify image instead

Flag looks good. I mostly am trying to get you to think about a plan for upgrading the default in the future. What usually happens is:
- customers start using it, say at version=ubuntu20
- we decide we want to change the default to ubuntu22 because ubuntu20 is EOL or something
- customers are unitentionally broken because the default changed and they depended on something ubuntu20 specific

Maybe we could make the default in the CLI "latest" or something and then decide that on the server? that way nobody is misled into thinking it will always be ubuntu20?

**Comment by maggie-lou** [comment]
> > > I don't think anyone is using bazel remote atm so this seems fine, assuming that's all it will affect.
> > > Would it make sense to have remote bazel use the defaults from the buildbuddy.yaml file in the repository? (and fall back to the default buildbuddy.yaml if there is none)
> > 
> > 
> > People can specify a container image for each action in buildbuddy.yaml, so it might get complicated to pick a single default. I added a flag to remote bazel to specify image instead
> 
> Flag looks good. I mostly am trying to get you to think about a plan for upgrading the default in the future. What usually happens is:
> 
> * customers start using it, say at version=ubuntu20
> * we decide we want to change the default to ubuntu22 because ubuntu20 is EOL or something
> * customers are unitentionally broken because the default changed and they depended on something ubuntu20 specific
> 
> Maybe we could make the default in the CLI "latest" or something and then decide what version that is on the server? that way nobody is misled into thinking it will always be ubuntu20?

Are you thinking if someone reads the CLI code and sees ubuntu20 and makes assumptions about that? I changed the flag default to be "", so we don't set a default runner image until hostedrunner.go. Is that what you mean?

We'd have to notify customers if we upgrade the default image in hostedrunner.go, but that feels inevitable

File: cli/remotebazel/remotebazel.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ const (
 var (
 	execOs            = flag.String("os", "linux", "If set, requests execution on a specific OS.")
 	execArch          = flag.String("arch", "amd64", "If set, requests execution on a specific CPU architecture.")
+	runnerImage       = flag.String("runner_image", "", "If set, requests execution on a specific runner image. Otherwise uses the default hosted runner version.")
```
Comment (bduffany) [inline cli/remotebazel/remotebazel.go:57]: Should we call this `container_image` for better consistency with the `container-image` exec property + `buildbuddy.yaml` config field?

File: proto/runner.proto [hunk 2]
```diff
@@ -83,6 +83,9 @@ message RunRequest {
 
   // If true, start the runner but do not wait for the action to be scheduled.
   bool async = 9;
+
+  // Container image to run on. Defaults to the Ubuntu 20.04 Workflows image.
+  string image = 10;
```
Comment (bduffany) [inline proto/runner.proto:88]: `container_image` here too, to match the exec property / bb.yaml name?

Comment (bduffany) [reply proto/runner.proto:88]: also document that a `docker://` suffix is required?

File: cli/remotebazel/remotebazel.go (hunk 1 above)
Comment (bduffany) [reply cli/remotebazel/remotebazel.go:57]: This was marked resolved - forgot to push?

File: cli/remotebazel/remotebazel.go [hunk 3]
```diff
@@ -54,6 +54,7 @@ const (
 var (
 	execOs            = flag.String("os", "linux", "If set, requests execution on a specific OS.")
 	execArch          = flag.String("arch", "amd64", "If set, requests execution on a specific CPU architecture.")
+	containerImage    = flag.String("container_image", "", "If set, requests execution on a specific runner image. Otherwise uses the default hosted runner version.")
```
Comment (bduffany) [inline cli/remotebazel/remotebazel.go:57]: nit: add to the help description that docker/OCI images require a `docker://` prefix

**Review by bduffany — APPROVED** [review]

**Pull request merged**